src/business_logic/execution/execution_entry_point.rs

```diff
@@ -122,7 +126,7 @@ impl ExecutionEntryPoint {
         let mut cairo_runner = CairoRunner::new(&contract_class.program, "all", false)?;
         cairo_runner.initialize_function_runner(&mut vm)?;
 
-        let mut tmp_state = T::default();
```

Comment:
I am not sure of this line.
Since we won't always want to use a CachedState with an InMemoryStateReader.

Reply:
I definitely don't understand the underlying `StarknetRunner` code good enough, but to me it seems that this `tmp_state` is not used _at all_ and is required just because some kind of API deficiencies so that we can fetch the initial `syscall_ptr`.

In fact, if you substitute the `CachedState(InMemoryStateReader)` instance here with a type "implementing" both `State` and `StateReader` via having `unimplemented!` in each trait method all tests just continue working fine.

I used `CachedState(InMemoryStateReader)` because that was the only implementation in the crate that was sufficient here (and as demonstrated by the experiment above it's not actually used either).

Reply:
Interesting. @juanbono it may be a good idea to review that API.

Reply:
I do agree with @pefontana tho, hardcoding it is not a good solution to broken APIs.

Reply:
I've made some changes here: if we move  `prepare_os_context()` out to a standalone function we can avoid the whole chicken-or-the-egg situation of having to construct a hint processor to get all the required data to construct a proper hint processor.